### PR TITLE
Apply ShutterCount Normalisation Correction when ShutterCounts Available

### LINKED ISSUE #2095: Spectrum Viewer: AutoScale Sample Against Open-Beam if Shutter Count Files present for Selected Stacks [Part 2]
### Desired Behaviour

IMAT have asked if we could use the shutter count files for each stack to autoscale sample against open-beam. 

Using the shutter count files for autoscaling will achieved more accurate results where counts across stacks differ

We should perform a check to within the spectrum viewer model to validate if shutter count files exist for the selected sample and open beam stacks, and autoscale using shutter count files if present. 

### Current Behaviour

Autoscaling using the shutter count files is not yet possible.

### PR DESCRIPTION
**Pull request by JackEAllen**
### Issue

Closes #2095 

### Description

*Add a description of the changes made*.
* Load ShutterCount Files into Spectrum Viewer if available.
* Only apply ShutterCount correction if bool propagated from view (currently not fully integrated)
* Apply ShutterCount correction to normalisation using only first value, but written such that more values can be used later on should requirements change.

### Testing 

*Describe the tests that were used to verify your changes*.
- Manually tested through hard coding normalise parameter in `get_spectrum` to True and adding log statements
- Visual inspection of how spectrum differed based on whether ShutterCount correction has been applied or not.
- Unit tests pass

### Acceptance Criteria 

*How should the reviewer test your changes*?
- Unit tests pass
- Through adding additional logs and viewing spectrum, hard coding `get_spectrum` `normalise` param to "True" applies ShutterCount correction

### Documentation

*How have you changed the documentation to reflect your changes? All changes should be noted in the appropriate file in docs/release_notes*

`docs/release_notes/next/feature-2095-apply_shuttercounts_normalisation`

### CODE REVIEW

**Review by samtygier-stfc — CHANGES_REQUESTED** [review]

File: mantidimaging/gui/windows/spectrum_viewer/model.py [hunk 1]
```diff
@@ -228,12 +228,34 @@ def get_spectrum(self, roi: str | SensibleROI, mode: SpecType) -> np.ndarray:
...
+        values should we find the initial value is not sufficient.
+        """
+        sample_shuttercount = self.get_stack_shuttercounts(self._stack)
+        open_shuttercount = self.get_stack_shuttercounts(self._normalise_stack)
+        if sample_shuttercount is None or open_shuttercount is None:
+            return 1.0  # No shutter count data available so no correction needed
+        normalised_shuttercounts = sample_shuttercount / open_shuttercount
+        return normalised_shuttercounts[0]
+
+    def get_transmission_error_standard_dev(self, roi: SensibleROI, normalised: bool = True) -> np.ndarray:
```
[the remark this answers is not in the document]
Comment (JackEAllen) [reply mantidimaging/gui/windows/spectrum_viewer/model.py:254]: Thanks for spotting this. This was left set to false for testing

File: mantidimaging/gui/windows/spectrum_viewer/model.py [hunk 2]
```diff
@@ -242,19 +278,30 @@ def get_transmission_error_standard_dev(self, roi: SensibleROI) -> np.ndarray:
...
+        if normalise_with_shuttercount:
+            average_shuttercount = self.get_shuttercount_normalised_correction_parameter()
+            safe_divide = safe_divide / average_shuttercount
+
         return np.std(safe_divide, axis=(1, 2))
 
-    def get_transmission_error_propagated(self, roi: SensibleROI) -> np.ndarray:
+    def get_transmission_error_propagated(self,
+                                          roi: SensibleROI,
+                                          normalise_with_shuttercount: bool = True) -> np.ndarray:
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/spectrum_viewer/model.py:289]: This one should also be false

**Review by samtygier-stfc — CHANGES_REQUESTED** [review]

File: mantidimaging/gui/windows/spectrum_viewer/model.py [hunk 3]
```diff
@@ -228,12 +231,45 @@ def get_spectrum(self, roi: str | SensibleROI, mode: SpecType) -> np.ndarray:
                 return np.array([])
             roi_spectrum = self.get_stack_spectrum(self._stack, roi)
             roi_norm_spectrum = self.get_stack_spectrum(self._normalise_stack, roi)
-        return np.divide(roi_spectrum, roi_norm_spectrum, out=np.zeros_like(roi_spectrum), where=roi_norm_spectrum != 0)
+            average_shuttercount = self.get_shuttercount_normalised_correction_parameter()
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/spectrum_viewer/model.py:234]: This is a stray line, its not needed as there  the same at 240 in the if statement

**Review by samtygier-stfc — APPROVED** [review]
Code looks good. I have tested with a few extra print statements to see the normalisation happening.

**Pull request merged**